tests/test_utils.py

```diff
@@ -39,8 +41,68 @@ def synapse_store():
         synapse_store = SynapseStorage()
     yield synapse_store
 
-
 class TestGeneral:
+    def test_clear_synapse_cache(self, tmp_path):
+        # define location of mock synapse cache
+        mock_synapse_cache_dir = tmp_path / ".synapseCache/"
+        mock_synapse_cache_dir.mkdir()
+        mock_sub_folder = mock_synapse_cache_dir / "123"
+        mock_sub_folder.mkdir()
+        mock_table_query_folder = mock_sub_folder/ "456"
+        mock_table_query_folder.mkdir()
+
+        # create mock table query csv and a mock cache map
+        mock_synapse_table_query_csv = mock_table_query_folder/ "mock_synapse_table_query.csv"
+        mock_synapse_table_query_csv.write_text("mock table query content")
+        mock_cache_map = mock_table_query_folder/ ".cacheMap"
+        mock_cache_map.write_text(f"{mock_synapse_table_query_csv}: '2022-06-13T19:24:27.000Z'")
+
+        assert os.path.exists(mock_synapse_table_query_csv)
+
+        # since synapse python client would compare last modified date and before date
+        # we have to create a little time gap here
+        time.sleep(1)
+
+        # clear cache
+        my_cache = cache.Cache(cache_root_dir=mock_synapse_cache_dir)
+        clear_synapse_cache(my_cache, minutes=0.0001)
+        # make sure that cache files are now gone
+        assert os.path.exists(mock_synapse_table_query_csv) == False
+        assert os.path.exists(mock_cache_map) == False
+    
+    def test_calculate_datetime_before_minutes(self):
+        input_date = datetime.strptime("07/20/23 17:36:34", '%m/%d/%y %H:%M:%S')
+        minutes_before = calculate_datetime(input_date=input_date, minutes=10, before_or_after="before")
+        expected_result_date_before = datetime.strptime("07/20/23 17:26:34", '%m/%d/%y %H:%M:%S')
+        assert minutes_before == expected_result_date_before
+
+    def test_calculate_datetime_after_minutes(self):
+        input_date = datetime.strptime("07/20/23 17:36:34", '%m/%d/%y %H:%M:%S')
+        minutes_after = calculate_datetime(input_date=input_date, minutes=10, before_or_after="after")
+        expected_result_date_after = datetime.strptime("07/20/23 17:46:34", '%m/%d/%y %H:%M:%S')
+        assert minutes_after == expected_result_date_after
+
+    def test_calculate_datetime_raise_error(self):
+        with pytest.raises(ValueError):
+            input_date = datetime.strptime("07/20/23 17:36:34", '%m/%d/%y %H:%M:%S')
+            minutes = calculate_datetime(input_date=input_date, minutes=10, before_or_after="error")
+    
+        
+    def test_check_synapse_cache_size(self,tmp_path):
```

Comment:
Just add a comment so that users using windows won't be surprised that the test fail. Or add pytest mark to mark this test and skip this test on windows.

Reply:
Yes sure. I could add that